File: Server/Server/Server/Model/DTO/RegisterUserDTO.cs
namespace Server.Model.DTO
{
    /// <summary>
    /// Représente les données d'inscription d'un utilisateur.
    /// </summary>
    public class RegisterUserDTO
    {
        private string username;
        private string email;
        private string password;
        private IFormFile? profilePic;

        /// <summary>
        /// Le nom de l'utilisateur souhaitant s'inscrire
        /// </summary>
        public string Username { get => username; set => username = value; }

        /// <summary>
        /// Le mail de l'utilisateur souhaitant s'inscrire
        /// </summary>
        public string Email { get => email; set => email = value; }

        /// <summary>
        /// Le mot de passe de l'utilisateur souhaitant s'inscrire
        /// </summary>
        public string Password { get => password; set => password = value; }

        /// <summary>
        /// La photo de profil de l'utilisateur souhaitant s'inscrire
        /// </summary>
        public IFormFile? ProfilePic { get => profilePic; set => profilePic = value; }
    }
}

File: Client/src/app/Model/DTO/RegisterUserDTO.ts
/**
 * Gère les attributs de l'objet d'inscription utilisateur (DTO = Data Transfer Object) sert d'objet de transfert de 
 * données entre un client et un serveur
 */
export class RegisterUserDTO {
  private username: string;
  private email: string;
  private password: string;
  private profilePic: File | null;

  /**
   * Renvoi le nom de l'utilisateur
   */
  public get ProfilePic(): File | null {
    return this.profilePic;
  }

  /**
   * Renvoi le mot de passe de l'utilisateur
   */
  public get Password(): string {
    return this.password;
  }

  /**
   * Renvoi l'email de l'utilisateur
   */
  public get Email(): string {
    return this.email;
  }

  /**
   * Renvoi le nom de l'utilisateur
   */
  public get Username(): string {
    return this.username;
  }

  constructor(username: string, email: string, password: string, profilePic: File | null) {
    this.username = username;
    this.email = email;
    this.password = password;
    this.profilePic = profilePic;
  }
}
